packages/smooth_app/lib/background/background_task_hunger_games.dart

```diff
@@ -0,0 +1,164 @@
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/background/abstract_background_task.dart';
+import 'package:smooth_app/data_models/operation_type.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/helpers/robotoff_insight_helper.dart';
+import 'package:smooth_app/query/product_query.dart';
+
+/// Background task about answering a hunger games question.
+class BackgroundTaskHungerGames extends AbstractBackgroundTask {
+  const BackgroundTaskHungerGames._({
+    required super.processName,
+    required super.uniqueId,
+    required super.barcode,
+    required super.languageCode,
+    required super.user,
+    required super.country,
+    required super.stamp,
+    required this.insightId,
+    required this.insightAnnotation,
+  });
+
+  BackgroundTaskHungerGames._fromJson(Map<String, dynamic> json)
+      : this._(
+          processName: json['processName'] as String,
+          uniqueId: json['uniqueId'] as String,
+          barcode: json['barcode'] as String,
+          languageCode: json['languageCode'] as String,
+          user: json['user'] as String,
+          country: json['country'] as String,
+          stamp: json['stamp'] as String,
+          insightId: json['insightId'] as String,
+          insightAnnotation: json['insightAnnotation'] as int,
+        );
+
+  /// Task ID.
+  static const String _PROCESS_NAME = 'HUNGER_GAMES';
+
+  static const OperationType _operationType = OperationType.hungerGames;
+
+  final String insightId;
+  final int insightAnnotation;
+
+  @override
+  Map<String, dynamic> toJson() => <String, dynamic>{
+        'processName': processName,
+        'uniqueId': uniqueId,
+        'barcode': barcode,
+        'languageCode': languageCode,
+        'user': user,
+        'country': country,
+        'stamp': stamp,
+        'insightId': insightId,
+        'insightAnnotation': insightAnnotation,
+      };
+
+  /// Returns the deserialized background task if possible, or null.
+  static AbstractBackgroundTask? fromJson(final Map<String, dynamic> map) {
+    try {
+      final AbstractBackgroundTask result =
+          BackgroundTaskHungerGames._fromJson(map);
+      if (result.processName == _PROCESS_NAME) {
+        return result;
+      }
+    } catch (e) {
+      //
```

Comment:
I think there's something missing here :)

Reply:
> I think there's something missing here :)

Actually, nothing's missing here. We just try to fromJson a map, and very often it's going to fail because some expected fields won't be there - if it is a background task from a different class. All other background tasks work on the same principle. cf. `AbstractBackgroundTask.fromJson`.
That said, I agree that it's not very elegant.
Feel free to create an issue to refactor that `fromJson`.